src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -623,8 +624,14 @@ void GoogleCalendarSyncAdaptor::purgeDataForOldAccount(int oldId, SocialNetworkS
 void GoogleCalendarSyncAdaptor::beginSync(int accountId, const QString &accessToken)
 {
     SOCIALD_LOG_DEBUG("beginning Calendar sync for Google, account" << accountId);
-
     bool needCleanSync = !wasLastSyncSuccessful(accountId);
+    if (needCleanSync) {
+        // we do the cleanup here to ensure atomic deletion + storage.save()
+        SOCIALD_LOG_INFO("performing purge due to requiring clean sync for account:" << accountId);
+        purgeDataForOldAccount(accountId, SocialNetworkSyncAdaptor::CleanUpPurge);
+        m_storageNeedsSave = false;
```

Comment:
For m_storageNeedsSave, wouldn't it be simpler if finalCleanup() (+ctor) would be the only place setting it false after save(), elsewhere setting it true if changes are made?